### PR TITLE
Update docs markup

### PR DESCRIPTION
**Pull request by outoftardis**
# Description

- Introduced minor changes to the markup to fix layout issues
- Added captions for tables + additional layout info for pdf output
- Changed image directive to figure directive

### CODE REVIEW

File: docs/Makefile [hunk 1]
```diff
@@ -43,7 +43,7 @@ parse-doxygen: doxygen
 # Catch-all target: route all unknown targets to Sphinx using the new
 # "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
 html: create_rst_examples Makefile
-	@$(SPHINXBUILD) -W --keep-going -w docbuild-log.txt -n -t use_intelname -b $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
+	@$(SPHINXBUILD) -n -t use_intelname -b $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```
Comment (outoftardis) [inline docs/Makefile:46]:
```suggestion
	@$(SPHINXBUILD) -W --keep-going -w docbuild-log.txt -n -t use_intelname -b $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```

File: docs/source/daal/usage/algorithms.rst [hunk 2]
```diff
@@ -34,9 +34,12 @@ In computation modes that permit multiple calls to the ``compute()`` method,
...
-.. list-table::
+.. tabularcolumns::  |\Y{0.2}|\Y{0.8}|
+
+.. list-table:: Algorithm Input
    :widths: 10 60
    :align: left
+   :class: longtable
 
    * - ``input.set(Input ID, InputData)``
      - Use to set a pointer to the input argument with the ``Input ID`` identifier.
```
Comment (mvincerx) [inline docs/source/daal/usage/algorithms.rst:45]: There are illegal line breaks in the bulleted items below the list-table on this page: \docs\source\daal\usage\algorithms.rst:

Please correct them. They prevent the LaTeX build. Thanks.

File: docs/source/daal/algorithms/outlier_detection/univariate.rst [hunk 3]
```diff
@@ -81,9 +84,12 @@ For more details, see :ref:`algorithms`.
     If you do not provide at least one of the ``location``, ``scatter``, ``threshold`` inputs,
     the library will initialize all of them with the following default values:
 
-    .. list-table::
+    .. tabularcolumns::  |\Y{0.3}|\Y{0.7}|
```
Comment (mvincerx) [inline docs/source/daal/algorithms/outlier_detection/univariate.rst:87]: As discussed, I recommend NOT embedding `list-table` directive inside any other directive. 
Instead, keep syntax simple, like so, by changing  the `note` to a paragraph: 

```

If you do not provide at least one of the ``location``,
``scatter``, ``threshold`` inputs,
the library will initialize all of them with the following default
values:


.. tabularcolumns::  |\Y{0.3}|\Y{0.7}|

.. list-table:: Default Values for Algorithm Input of Univariate Outlier Detection (Batch Processing)
    :widths: 10 20
    :class: longtable

    * - ``location``
      - A set of :math:`0.0`

    * - ``scatter``
      - A set of :math:`1.0`

    * - ``threshold``
```

In general, try not to embed directives within directives. If done, it increases the probability of error w/ LaTeX builder.

**Review by mvincerx — COMMENTED** [review]
More comments will follow.

**Review by napetrov — APPROVED** [review]

**Pull request merged**